src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -296,6 +296,38 @@ public bool HasLaunchOption
             }
         }
 
+        public bool NativeCodeDebugging
+        {
+            get
+            {
+                if (!IsProfileSelected)
+                {
+                    return false;
+                }
+
+                if (SelectedDebugProfile.OtherSettings.Keys.Contains("nativeDebugging"))
+                {
+                    return (bool) SelectedDebugProfile.OtherSettings["nativeDebugging"];
+                }
+
+                return false;
+            }
+            set
+            {
+                //Unlike other properties that have default values, nativeDebugging may not be set yet. Because false is the default behavior adding it has no affect
+                if (!SelectedDebugProfile.OtherSettings.ContainsKey("nativeDebugging"))
+                {
+                    SelectedDebugProfile.OtherSettings.Add("nativeDebugging", false);
+                }
+
+                if (SelectedDebugProfile != null && (bool) SelectedDebugProfile.OtherSettings["nativeDebugging"] != value)
```

Comment:
It seems simpler if you just write the following, also avoids "dirtying" the in memory model when the value is set to false and it's not present in the settings:

``` C#
            SelectedDebugProfile.OtherSettings.TryGetValue("nativeDebugging", out object previousValue);

            if ((bool)previousValue != value)
            {
                SelectedDebugProfile.OtherSettings["nativeDebugging"] = value;
                OnPropertyChanged(nameof(NativeCodeDebugging));
            }
```

Reply:
Good point - changed